src/neuralnet/voting/builders.cpp

```diff
@@ -0,0 +1,1142 @@
+#include "init.h"
+#include "main.h"
+#include "neuralnet/beacon.h"
+#include "neuralnet/contract/contract.h"
+#include "neuralnet/contract/message.h"
+#include "neuralnet/researcher.h"
+#include "neuralnet/voting/builders.h"
+#include "neuralnet/voting/claims.h"
+#include "neuralnet/voting/payloads.h"
+#include "ui_interface.h"
+#include "wallet/wallet.h"
+
+#include <boost/algorithm/string/trim.hpp>
+
+using namespace NN;
+using LogFlags = BCLog::LogFlags;
+
+namespace {
+//!
+//! \brief Get the string representation of an output destination for logging.
+//!
+//! \param dest Destination to convert to an address string.
+//!
+//! \return Base58-encoded string of the address.
+//!
+std::string DestinationToAddressString(const CTxDestination dest)
+{
+    CBitcoinAddress address;
+    address.Set(dest);
+
+    return address.ToString();
+}
+
+//!
+//! \brief A set of outputs associated with an address for a voting claim.
+//!
+//! This intermediate container holds context about the unspent outputs for
+//! an address in the wallet to build voting claims from.
+//!
+class AddressOutputs
+{
+public:
+    CKeyID m_key_id;                    //!< Address of the outputs.
+    std::vector<COutPoint> m_outpoints; //!< Outputs for the address.
+    std::vector<int64_t> m_amounts;     //!< Amounts for each output.
+    int64_t m_total_amount;             //!< Total amount of the outputs.
+
+    //!
+    //! \brief Initialize an output address grouping.
+    //!
+    //! \param key_id The address that groups the contained outputs.
+    //!
+    AddressOutputs(const CKeyID key_id) : m_key_id(key_id), m_total_amount(0)
+    {
+    }
+
+    //!
+    //! \brief Determine whether the total amount of the grouped outputs
+    //! compares greater than than another address grouping.
+    //!
+    bool operator>(const AddressOutputs& other) const
+    {
+        return m_total_amount > other.m_total_amount;
+    }
+
+    //!
+    //! \brief Add an output to the address grouping.
+    //!
+    //! \param output An output associated with the address.
+    //!
+    void Add(const COutput& output)
+    {
+        m_outpoints.emplace_back(output.tx->GetHash(), output.i);
+        m_amounts.emplace_back(output.tx->vout[output.i].nValue);
+
+        m_total_amount += m_amounts.back();
+    }
+}; // AddressOutputs
+
+using OutputAddressSet = std::vector<AddressOutputs>;
```

Comment:
Is it a bit misleading to call this OutputAddressSet? Maybe OutputAddressVector?

Reply:
It doesn't matter codewise, of course... but some people might think this is an actual std::set without reviewing the above.

Reply:
That's true. I'm in the habit of using "map" as a generic alias for associative containers and "set" as an alias for non-associative containers when the type of the underlying container is an implementation detail, but that doesn't carry over to C++ very well. How about `OutputAddresses`?

Reply:
Ok

Reply:
(BTW I often use terms like set loosely in my comments too... and always fret about it wondering whether it will confuse people.)

Reply:
Naming is hard 🙂

Reply:
Yep.